src/main/java/org/folio/service/pieces/flows/update/PieceUpdateFlowManager.java

```diff
@@ -92,10 +108,49 @@ public Future<Void> updatePiece(Piece pieceToUpdate, boolean createItem, boolean
   }
 
   protected Future<Void> updatePoLine(PieceUpdateHolder holder, RequestContext requestContext) {
-    if (!Boolean.TRUE.equals(holder.getOriginPoLine().getIsPackage()) && !Boolean.TRUE.equals(holder.getOriginPoLine().getCheckinItems())) {
-      return updatePoLineService.updatePoLine(holder, requestContext);
-    }
-    return Future.succeededFuture();
+    CompositePoLine originPoLine = holder.getOriginPoLine();
+
+    return getPiecesByPoLine(originPoLine.getId(), requestContext)
+      .compose(pieces -> {
+        CompositePurchaseOrder order = holder.getOriginPurchaseOrder();
+        if (order.getOrderType() != OrderType.ONE_TIME || order.getWorkflowStatus() != WorkflowStatus.OPEN) {
+          return Future.succeededFuture();
+        }
+        logger.info(">>> " + pieces);
+        List<Piece> piecesToUpdate = List.of(holder.getPieceToUpdate());
+        logger.info(">>> " + calculatePoLineReceiptStatus(originPoLine, pieces, piecesToUpdate));
+        return Future.succeededFuture();
+      }).compose(t -> {
+        if (!Boolean.TRUE.equals(originPoLine.getIsPackage()) &&
+          !Boolean.TRUE.equals(originPoLine.getCheckinItems())) {
+          return updatePoLineService.updatePoLine(holder, requestContext);
+        }
+        return Future.succeededFuture();
+      });
+  }
+
+  PoLine.ReceiptStatus calculatePoLineReceiptStatus(CompositePoLine poLine, List<Piece> fromStorage, List<Piece> toUpdate) {
```

Comment:
the result value of this method is not used anywhere except log message and tests